src/features/view-subdomains/SubdomainTable/SubdomainTable.test.tsx

```diff
@@ -0,0 +1,214 @@
+}));
+
+const mockSubdomains = {
+	loadedSubdomains: {
+		pages: [
+			[
+				{
+					id: '0x123',
+					name: 'mock-1',
+				},
+				{
+					id: '0x234',
+					name: 'mock-2',
+				},
+			],
+			[
+				{
+					id: '0x345',
+					name: 'mock-3',
+				},
+			],
+		],
+	},
+	isLoading: false,
+};
+
+////////////////////////
+// Mock Subcomponents //
+////////////////////////
+
+jest.mock('../SubdomainTableCard', () => ({
+	SubdomainTableCard: (props) => <div data-testid="card">{props.zna}</div>,
+}));
+
+jest.mock('../SubdomainTableRow', () => ({
+	SubdomainTableRow: (props) => (
+		<tr data-testid="row">
+			<td>{props.zna}</td>
+		</tr>
+	),
+}));
+
+beforeEach(() => {
+	jest.resetAllMocks();
+});
+
+///////////
+// Tests //
+///////////
+
+describe('<SubdomainTable />', () => {
+	beforeEach(() => {
+		mockSubdomainSearch = mockSearchResult;
+		mockInfiniteSubdomains = mockSubdomains;
+	});
+
+	test('should render scroll trigger', () => {
+		const { container } = render(<SubdomainTable zna="0x123" />);
+		expect(container.getElementsByClassName('Trigger')).toHaveLength(1);
+	});
+
+	describe('when browsing subdomains', () => {
+		describe('when toggling between grid and table view', () => {
+			test('should be grid view by default', () => {
+				const { getAllByTestId } = render(<SubdomainTable zna="0x123" />);
+				expect(getAllByTestId('card').length).toBeTruthy();
+			});
+
+			test('should be able to toggle between grid and list views', () => {
+				/*
+				 * @note: this test is flaky, as it relies on the order of the radio buttons.
+				 * Improve this in future iterations!
+				 */
+
+				const { getAllByRole, queryByTestId, getAllByTestId } = render(
+					<SubdomainTable zna="0x123" />,
+				);
+
+				const [listViewButton, gridViewButton] = getAllByRole('radio');
+
+				fireEvent.click(listViewButton);
+
+				expect(getAllByTestId('row').length).toBeTruthy();
+				expect(queryByTestId('card')).toBeNull();
+
+				fireEvent.click(gridViewButton);
+
+				expect(getAllByTestId('card')).toBeTruthy();
+				expect(queryByTestId('row')).toBeNull();
+			});
+		});
+
+		describe('when loading initial subdomains', () => {
+			test('should display correct message: "Loading subdomains"', () => {
+				mockInfiniteSubdomains = {
+					...mockSubdomains,
+					isLoading: true,
+				};
+				const { getByText } = render(<SubdomainTable zna="0x123" />);
+				expect(getByText('Loading subdomains')).toBeInTheDocument();
+			});
+		});
+
+		describe('when loading next page of subdomains', () => {
+			beforeEach(() => {
+				mockInfiniteSubdomains = {
+					...mockSubdomains,
+					isLoading: true,
+					isFetchingNextPage: true,
+				};
+			});
+
+			test('should display correct message: "Loading more subdomains"', () => {
+				const { getByText } = render(<SubdomainTable zna="0x123" />);
+				expect(getByText('Loading more subdomains')).toBeInTheDocument();
+			});
+
+			test('should show already loaded subdomains', () => {
```

Comment:
~Looks like this test and the test on line 151 are duplicates from what I can see, could remove this one and keep the one below in the `when subdomains have loaded` describe block~

Edit: spotted the test set up is different - looks good !